book-content/chapters/08-classes.md

```diff
@@ -0,0 +1,1044 @@
+  title: "Loop Finding Jazz Records",
+  artist: "Jan Jelinek",
+  releaseYear: 2001,
+});
+
+console.log(loopFindingJazzRecords.title); // Output: Loop Finding Jazz Records
+
+const unknownAlbum = new Album();
+
+console.log(unknownAlbum.title); // Output: Unknown Album
+```
+
+### Using a Class as a Type
+
+An interesting property of classes in TypeScript is that they can be used as types for variables and function parameters. The syntax is similar to how you would use any other type or interface.
+
+In this case, we'll use the `Album` class to type the `album` parameter of a `printAlbumInfo` function:
+
+```typescript
+function printAlbumInfo(album: Album) {
+  console.log(
+    `${album.title} by ${album.artist}, released in ${album.releaseYear}.`,
+  );
+}
+```
+
+We can then call the function and pass in an instance of the `Album` class:
+
+```typescript
+printAlbumInfo(sixtyNineLoveSongsAlbum);
+
+// Output: 69 Love Songs by The Magnetic Fields, released in 1999.
+```
+
+While using a class as a type is possible, it's a much more common pattern to require classes to implement a specific interface.
+
+## Properties in Classes
+
+Now that we've seen how to create a class and create new instances of it, let's look a bit closer at how properties work.
+
+### Class Property Initializers
+
+You can set default values for properties directly in the class body. These are called class property initializers.
+
+```typescript
+class Album {
+  title = "Unknown Album";
+  artist = "Unknown Artist";
+  releaseYear = 0;
+}
+```
+
+You can combine them with type annotations:
+
+```typescript
+class Album {
+  title: string = "Unknown Album";
+  artist: string = "Unknown Artist";
+  releaseYear: number = 0;
+}
+```
+
+Importantly, class property initializers are resolved _before_ the constructor is called. This means you can override the default values by assigning a different value in the constructor:
+
+```typescript
+class User {
+  name = "Unknown User";
+
+  constructor() {
+    this.name = "Matt Pocock";
+  }
+}
+
+const user = new User();
+
+console.log(user.name); // Output: Matt Pocock
+```
+
+### `readonly` Class Properties
+
+As we've seen with types and interfaces, the `readonly` keyword can be used to make a property immutable. This means that once the property is set, it cannot be changed:
+
+```typescript
+class Album {
+  readonly title: string;
+  readonly artist: string;
+  readonly releaseYear: number;
+}
+```
+
+### Optional Class Properties
+
+We can also mark properties as optional in the same way as objects, using the `?:` annotation:
+
+```typescript
+class Album {
+  title?: string;
+  artist?: string;
+  releaseYear?: number;
+}
+```
+
+As we can see from the lack of errors above, this also means they don't need to be set in the constructor.
+
+### `public` and `private` properties
+
+The `public` and `private` keywords are used to control the visibility and accessibility of class properties.
```

Comment:
calling out both visibility and accessibility here calls out for defining both (or at least for differentiating between both). It's kinda the same thing here so I think I'd avoid using both terms and stick to one